src/main/java/fi/metatavu/metamind/images/ImageServlet.java

```diff
@@ -0,0 +1,170 @@
+
+import fi.metatavu.metamind.persistence.dao.KnotDAO;
+import fi.metatavu.metamind.persistence.models.Knot;
+
+
+@RequestScoped
+@MultipartConfig
+@WebServlet (urlPatterns = { "/v2/images", "/v2/images/*" })
+public class ImageServlet extends HttpServlet{
+
+  private static final long serialVersionUID = -5737983422302604984L;
+  @Inject
+  private Logger logger;
+  
+  @Inject 
+  private KnotDAO knotDAO;
+  
+  private void setCorsHeaders(HttpServletResponse resp) {
+    resp.setHeader("Access-Control-Allow-Origin", "*");
+    resp.setHeader("Access-Control-Allow-Headers", "Origin, X-Requested-With, Content-Type, Accept");
+    resp.setHeader("Access-Control-Allow-Methods", "GET,PUT,POST,DELETE,OPTIONS");
+  }
+  @Override 
+  protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException{
+    setCorsHeaders(resp);
+    String fileURL = req.getPathInfo();
+    File file = new File("images/"+fileURL);
+    if(!file.exists()) {
+      resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    resp.setStatus(HttpServletResponse.SC_FOUND);
+    try {
+      ServletOutputStream servletOutputStream = resp.getOutputStream();
+      try (InputStream data = new FileInputStream(file)) {
+        IOUtils.copy(data, servletOutputStream);
+      } finally {
+        servletOutputStream.flush();
+      }
+    } catch (IOException e) {
+      logger.warn("Failed to send response", e);
+    }
+  }
+  
+  @Override 
+  protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException{
+    setCorsHeaders(resp);
+    String fileURL = req.getPathInfo();
+    File file = new File("images/"+fileURL);
+    if(!file.exists()) {
+      resp.setStatus(HttpServletResponse.SC_NOT_FOUND);
+      return;
+    }
+    file.delete();
+    resp.setStatus(HttpServletResponse.SC_OK);
+  }
+  
+  
+  @Override
+  protected void doPut(HttpServletRequest req, HttpServletResponse resp) throws ServletException, IOException {
+    
+    try {
+      Part file = req.getPart("file");
+      Part idPart = req.getPart("knotId");
+      String contentType = file.getContentType();
+      if ( file == null || idPart == null || !contentType.startsWith("image") ) {
+        resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      
+      setCorsHeaders(resp);
+      
+      InputStream knotIdInputStream = idPart.getInputStream();
+      UUID knotId = UUID.fromString(convertStreamToString(knotIdInputStream,Charset.defaultCharset()));
+      
+      Knot knot = knotDAO.findById(knotId);
+      
+      if ( knot == null ) {
+        resp.setStatus(HttpServletResponse.SC_BAD_REQUEST);
+        return;
+      }
+      
+      String extension = file.getSubmittedFileName().substring(file.getSubmittedFileName().lastIndexOf("."));
+      String fileURL = "images/"+knotId.toString()+extension;
+      InputStream fileInputStream = file.getInputStream();
+      
+      File dir = new File("images");
```

Comment:
Make folder configurable with for example system property

Reply:
Folder can now be configured using system property "metamind.imagedir"